Automattic/newspack-plugin
# UI fixes for Stripe setup in Reader Revenue wizard

### All Submissions:






### Changes proposed in this Pull Request:



- Refactors Stripe-related code in preparation for feature expansion. 
- Fixes some UI issues

### How to test the changes in this Pull Request:

1. On `master`, remove (or temp. rename) the `newspack_stripe_data` option (to simulate a new site – you can also use a fresh new site)
2. Visit the Reader Revenue wizard and choose Stripe as the platform 
3. Go to Stripe Settings tab and observe errors being displayed: 

<img width="785" alt="image" src="https://user-images.githubusercontent.com/7383192/202760803-9c126c10-7d02-4aa4-9dc4-7577665629c4.png">

4. Observe a `PHP Warning:  Attempt to read property "webhookEndpoints" on null` warning logged 
5. Switch to this branch, reload, observe no errors and no warning logged 
6. Back to `master`! Now, save Stripe Setting without inputting any settings. 
7. Observe that after request completes, the tabs are in a limbo state and the UI changes to platform selection:

<img width="890" alt="image" src="https://user-images.githubusercontent.com/7383192/203017959-4785c598-f51f-43d9-9901-2dc7036b2fee.png">

8. Switch to this branch, repeat, observe the error is displayed below the navigation tabs, and that the UI is not reset:

<img width="837" alt="image" src="https://user-images.githubusercontent.com/7383192/203018145-35a08886-4434-44f6-a6a7-b2e0d82533b0.png">

### Other information:
